packages/wallets/skywallet/src/adapter.ts

```diff
@@ -0,0 +1,248 @@
+    }
+
+    async connect(): Promise<void> {
+        try {
+            if (this.connected || this.connecting) return;
+            if (this._readyState !== WalletReadyState.Installed) throw new WalletNotReadyError();
+
+            this._connecting = true;
+
+            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+            const wallet = window!.skySolana!;
+            try {
+                await wallet.connect();
+            } catch (error: any) {
+                throw new WalletConnectionError(error?.message, error);
+            }
+
+            if (!wallet.publicKey) {
+                throw new WalletAccountError();
+            }
+
+            let publicKey: PublicKey;
+            try {
+                publicKey = new PublicKey(wallet.publicKey.toBytes());
+            } catch (error: any) {
+                throw new WalletPublicKeyError(error?.message, error);
+            }
+
+            wallet.on('disconnect', this._disconnected);
+
+            this._wallet = wallet;
+            this._publicKey = publicKey;
+
+            this.emit('connect', publicKey);
+        } catch (error: any) {
+            this.emit('error', error);
+            throw error;
+        } finally {
+            this._connecting = false;
+        }
+           
+    }
+
+    async disconnect(): Promise<void> {
+        const wallet = this._wallet;
+        if (wallet) {
+            wallet.off('disconnect', this._disconnected);
+
+            this._wallet = null;
+            this._publicKey = null;
+
+            try {
+                await wallet.disconnect();
+            } catch (error: any) {
+                this.emit('error', new WalletDisconnectionError(error?.message, error));
+            }
+        }
+
+        this.emit('disconnect');
+    }
+
+    async sendTransaction(
+        transaction: Transaction,
+        connection: Connection,
+        options: SendTransactionOptions = {}
+    ): Promise<TransactionSignature> {
+        try {
+            const wallet = this._wallet;
+            if (!wallet) throw new WalletNotConnectedError();
+            //SKY wallet will try to setup defaults in case user has not provided feePayer and blockhash values in the transaction
+            transaction.feePayer = transaction.feePayer || this.publicKey || undefined
+            transaction.recentBlockhash = transaction.recentBlockhash || (await connection.getRecentBlockhash('finalized')).blockhash;
+
+            if (options.signers?.length) {
+                // handling partial signing 
+                const { signers, ...sendOptions } = options;
+                transaction.partialSign(...signers);
+                transaction = await this.signTransaction(transaction);
+                const rawTransaction = transaction.serialize(); 
+  
+                const signature = await connection.sendRawTransaction(rawTransaction, sendOptions); 
+                return signature
+            }
```

Comment:
This doesn't need to be reimplemented, does it? The base class already has all this, you can call the super method if you don't support partial signers.